File: Data Structures & Algorithms/binary-tree-inorder-traversal/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        vector<int> output {};
        stack<TreeNode*> st {};
        TreeNode* current {root};
        while (current || !st.empty()) {
            if (current) {
                st.push(current);
                current = current->left;
            }
            else {
                current = st.top();
                st.pop();
                output.push_back(current->val);
                current = current->right;
            }
        }
        return output;
    }
};

File: Data Structures & Algorithms/evaluate-reverse-polish-notation/submission-1.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        output = 0
        stack = []
        for token in tokens:
            match token:
                case "+":
                    while stack:
                        output += int(stack.pop())
                case "-":
                    while stack:
                        output -= stack.pop()
                case "*":
                    while stack:
                        output *= stack.pop()
                case "/":
                    denom = 0
                    while not denom:
                        denom = stack.pop()
                    nom = stack.pop()
                    stack.append(nom // denom)
                case _:
                    stack.append(int(token))
        return output

                
        